File: solutions/go/bird-watcher/1/bird_watcher.go
package birdwatcher

// TotalBirdCount return the total bird count by summing
// the individual day's counts.
func TotalBirdCount(birdsPerDay []int) int {
	s := 0
    for _, v := range(birdsPerDay) {
        s += v
    }
    return s
}

// BirdsInWeek returns the total bird count by summing
// only the items belonging to the given week.
func BirdsInWeek(birdsPerDay []int, week int) int {
    s := 0
	for _, v := range(birdsPerDay[7 * (week-1):7 * week]) {
        s += v
    }
    return s
}

// FixBirdCountLog returns the bird counts after correcting
// the bird counts for alternate days.
func FixBirdCountLog(birdsPerDay []int) []int {
	for i := range(birdsPerDay) {
        if i % 2 == 1 {
            continue
        }
        birdsPerDay[i]++
    }
    return birdsPerDay
}

File: solutions/javascript/bird-watcher/3/bird-watcher.js
export function totalBirdCount(birdsPerDay) {
  return Array.prototype.reduce.call(birdsPerDay, (a, b) => a + b, 0)
}

export function birdsInWeek(birdsPerDay, week) {
  return Array.prototype.slice.call(birdsPerDay, 7 * (week-1), 7 * week).reduce((a, b) => a + b, 0)
}

export function fixBirdCountLog(birdsPerDay) {
  Array.prototype.forEach.call(birdsPerDay, (v, i, a) => a[i] = i % 2 === 0 ? v + 1 : v)
  return birdsPerDay
}